web/ui/react-app/src/hooks/useFetch.ts

```diff
@@ -86,33 +57,61 @@ export const useFetchReadyInterval = (pathPrefix: string, options?: RequestInit)
   const [walReplayStatus, setWALReplayStatus] = useState<WALReplayStatus>({} as any);
 
   useEffect(() => {
-    const interval = setInterval(async () => {
-      try {
-        let res = await fetch(`${pathPrefix}/-/ready`, { cache: 'no-store', credentials: 'same-origin', ...options });
-        if (res.status === 200) {
-          setReady(true);
-          clearInterval(interval);
-          return;
-        }
-        if (res.status !== 503) {
-          setIsUnexpected(true);
-          setWALReplayStatus({ data: { last: 0, first: 0 } } as any);
-        } else {
-          setIsUnexpected(false);
+    if (wasReady) {
+      setReady(true);
+    } else {
+      // This helps avoid a memory leak.
+      let mounted = true;
```

Comment:
Could you explain the leak a bit? As far as I can see, this is used to control whether the hook state setters are called on a mounted vs. unmounted component, but that shouldn't change anything with regards to actually leaking memory, it would just hide the memory leak warning, right? With the leak being in the form of the still-running `fetchStatus()`. Would the proper way be to cancel the `fetch()` request upon unmount? I found https://www.robinwieruch.de/react-warning-cant-call-setstate-on-an-unmounted-component which also presents your workaround, but AFAIU it's more of a warning suppressor than an actual memory leak preventor.

Reply:
So I tried using an abort signal to cancel the request and the error still presented itself, although this may require further investigation.

Reply:
Hmm, odd. Ok, let's keep it like this for now.